src/platform/user/authentication/components/SessionTimeoutModal.jsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import moment from 'moment';
+
+import Modal from '@department-of-veterans-affairs/formation-react/Modal';
+
+import recordEvent from 'platform/monitoring/record-event';
+import { logout } from 'platform/user/authentication/utilities';
+import { teardownProfileSession } from 'platform/user/profile/utilities';
+import localStorage from 'platform/utilities/storage/localStorage';
+
+const MODAL_DURATION = 30; // seconds
+
+class SessionTimeoutModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { countdown: null };
+    this.expirationInterval = null;
+  }
+
+  componentDidUpdate() {
+    if (this.props.isLoggedIn && !this.expirationInterval) {
+      this.clearInterval();
+      this.expirationInterval = setInterval(this.checkExpiration, 1000);
```

Comment:
Would there be any performance concerns about checking for expiration this frequently?  Could this maybe be done every 5 or 30 seconds?